tests/chainerx_tests/unit_tests/routines_tests/test_math.py

```diff
@@ -1098,67 +1098,51 @@ def forward_xp(self, inputs, xp):
 
 
 @op_utils.op_test(['native:0', 'cuda:0'])
-@pytest.mark.parametrize('input', [
-    numpy.asarray(0), numpy.asarray(-1), numpy.asarray(1), numpy.asarray(
-        10), numpy.full((), 2), numpy.full((0,), 2), numpy.full((2, 3), 2),
-    numpy.random.randn(2, 4), numpy.random.randn(4, 4),
-    numpy.random.randn(1, 4)
-])
-@pytest.mark.parametrize('contiguous', [None, 'C'])
-@pytest.mark.parametrize('in_dtype,out_dtype', _expected_dtypes_math_functions)
-class TestSin(op_utils.NumpyOpTest):
-
-    def setup(self, input, contiguous, in_dtype, out_dtype):
-        self.input = input.astype(in_dtype)
-        self.chx_dtype = out_dtype
-        self.contiguous = contiguous
-
-        if in_dtype == 'float16':
-            self.check_backward_options = {'atol': 5e-4, 'rtol': 5e-3}
-            self.check_double_backward_options = {'atol': 5e-3, 'rtol': 5e-2}
-        if numpy.dtype(in_dtype).kind != 'f':
-            self.skip_backward_test = True
-            self.skip_double_backward_test = True
-
-    def generate_inputs(self):
-        return self.input,
+@chainer.testing.parameterize(*(
+    # Special shapes
+    chainer.testing.product({
+        'shape': [(), (0,), (1,), (2, 0, 3), (1, 1, 1), (2, 3)],
+        'dtype': chainerx.testing.all_dtypes,
```

Comment:
Could you merge cuurent master branch and write as `'expected_dtypes': _expected_dtypes_math_functions,`?

Reply:
Sure on it.

Reply:
I ll rebase on the current master instead of merge. Hope that is okay.